### ISSUE TITLE
bangCall callback chain indentation bug

### ISSUE THREAD

**Issue by rattrayalex** [wcjohnson/lightscript#33]
see [lscdiag](https://wcjohnson.github.io/lscdiag/?v=1&s=eMOawrVRw4FOw4MwDMO9wpXDjMKnIcKVwqrDk8OUDirCgcKKwrjDswNjQmnDosK2QWlSUkcawqvDusOvJBsHLlRCaMK5JMO2e37DtnMmaMOUwpHCvMODEcOKCcOeej7DhMK7wqp6JC45w7EYwonDtzt5w5LCu8OiMwtnwqvCisOdwqnDicK7LsK3H1DCksOzwphcCD7Cl8O3AcOfRAIdw7sib8Orw65CwpjCkyXCiQlGw6vCncOAw7jDksKjwoAyw4Q9PglSw5bCrG9eDWMpdWhWTMKywodHFmPDhsOkOS04woluw4XDkMK5H8KIwo3DlcOUKcOTwq4DEMOqIcK0w7cqwqoTwq_Cg8OLfcKWbA4JwozCqFEQw4oXK0Nrw6PCtcKOY1rDk8Kow7bDrMOZw7bCg8OSw6jDgjhWwrXDjsO6wqHDmsKmWcKaw50ODiHCgQbDucO3w5LDtlBzw5M-c8Ktw6Eww4_Ci8K7w7jCm8OTf3rDjMKuw6YxaGg1UkzDv8O-wq1LwosIGl9KKcOKw5s#)

tl;dr this is broken:
```
someAction()
  .then! d => 
    d
  .catch! err => 
    dosomething(err)

**Comment by wcjohnson** [comment]
Yup, that's definitely wrong, good catch. I use `noEnforcedSubscriptIndentation` which I guess is why I haven't run into this one myself.

**Comment by wcjohnson** [comment]
**NOTE** any fix must be backported to 2.x

**Comment by wcjohnson** [comment]
Bisected this to a change in 2.3.1. Issue doesn't exist in 2.2: [lscdiag](https://wcjohnson.github.io/lscdiag/?v=1&s=eMOaw4VRw4FOw4MwDMO9wpXDjMKnIcKVwqrDm8OUDirCgcKKwrjDswNjQmnDosK2QWlSUkcawqvDusOvJBsHLlTDokQuwonDvcKewp_DvcKcCRp1IsOvcMKEcsKCwrfCng_DscKuwqoewolLTjxGw6LDvU7CnsO1wr7DuMOMw4LDmcKpYn9uw7LCrsOLw60HwpTDpDwmV8KCw4_DpX3DgDfCkUDCp8K-w4jDm8K6wrsSw6ZkSWLCgsOReicwwr7DtCjCoAxxwo9PwoLClDXDq8KbV8ODWEodwpoVwpPDrMOhwpHDhcKYMXlJC07ColsxdMOuB2JjNXXDisK0w6sAwoR6CMOtwr3CisOqw4TDq8Ogw7LCkCXDmTHCgRE1CkLDuWJlaG3CvMOWcUxrGsOVXjzDm35QGl0Yw4fCqsOWWT9Uw5t0wptuIMKBBsO5w7fDgg5Qc8OTPnPCrcOhOMOPwot7w7jCm8OLf8OxwrdLwrM0wrsdHC54DBpawo0Uw5PCv3_DqcOSIsKCw4YXw6xSw4lo)

**Comment by wcjohnson** [comment]
So, the problem here was the `.catch` subscript attaching to `d`, triggering the indentation error because it's not indented from `d`.

In 2.2, the bangCall unwinding algorithm crossed whiteblock boundaries, so it saw `d` as being a possible subscript for the bangCall and unwound the subscript chain by one level, in effect getting it right by accident.

In 2.3, I patched the bangCall unwinding algorithm to respect block boundaries, so the unwinding is no longer happening and the subscript is attaching to `d`. This carried over into the 3.0 alphas as well.

The correct answer, I believe, is that it doesn't make sense for subscripting to cross whiteblock boundaries at all, in any circumstances -- so that's what I've done here. I believe it should fix this issue.

**Comment by wcjohnson** [comment]
Fixed in 3.x and backported to 2.3. Closing.
